### DISCUSSION TITLE
Pumpenleistung, Mischerstellung und Ölverbrauch

### DISCUSSION THREAD

**Discussion by rheindt-sven** [the78mole/esphome_components#70] [Q&A] [answered]
Hi,
Vier Fragen:

1) sind die beiden Sensoren Pumpenleistung und Mischerstellung bei euch auch nur 0%, 100% oder -100%? Dann wäre das ja eher on/off.. 

2) kann/könnte man diese Parameter auch selbst steuern, oder kommt man da nicht dran?

3) ich habe keinen Ölverbrauchzähler.. welcher Parameter ist das beste Äquivalent dazu? "Brenneransteuerung = on"? Es gibt auch noch sowas wie "Kessel_unter_Betrieb", etc. 

4) mit welchen Paramtern bekommt man am sinnvollsten heraus wie viel Ölverbrauch/Brennerlaufzeit (ist ja nur aus oder Vollgas, oder?) in HK1, HK2 und WW geht?

Vielen Dank euch

**Comment by the78mole** [comment, marked as answer]
Hallo Sven,

bei 1) und 2) muss ich tatsächlich passen. Kann ich nicht ohne tiefere Recherche beantworten. Zu 2) sagt mein Bauchgefühl: Eher nicht,

zu 3), das geht bei nicht-modulierenden Brennern (wie ich und viele andere ihn haben) ganz einfach über den Betriebsstundenzähler, wenn man nicht erwartet, dass es auf den Milliliter genau ist. Die Ölmenge hängt nämich hauptsächlich von der verwendeten Düse des Brenners ab. Mein Brenner beispielsweise hat eine Düse verbaut, die etwas mehr als 2 l/h durchlässt. Wenn man nicht weiß, welche Düse verbaut ist, dann kann man das auch über die Kesselleistung grob rückrechnen (10 kW ~ 1 l/h). Den Ölstand im Tank messe ich absolut mit einem OilFox.

Da mich die Ölmenge selbst nicht so sehr interessiert, rechne ich direkt von der Kesselleistung über die Betriebsstunden auf kWh um. Da korrigiere ich dann noch mit 90% Wirkungsgrad (also nicht 21 kW Nennleistung des Brenners sondern 23 kW Ölverbrauch) und speise es als Energiezählwert in Home Assistant ein (als Gasverbrauch, weil es in HA keinen Ölverbrauch gibt),

Was die Aufteilung auf HK1, HK2 und WW angeht, muss man dann schon Wärmemengenzähler verbauen. Das bekommt man allenfalls sehr umständlich und sehr ungenau über die Aktivierungszeit der Ladepumpe und die Vorlauftemperaturen hin. Man könnte davon ausgehen, dass die HK nicht aktiv sind, wenn WW bereitet wird. Die Durchflussmenge bei WW ist konstant, aber die beim HK hängt von den Reglern an den Heizkörpern ab. Also alles sehr komplex zu verrechnen und mit vielen Fehlern behaftet.

Hoffe, das beantwortet einen Teil Deiner Fragen ausreichend...

Grüße,
Dein Maulwurf

**Comment by dewenni** [comment]
Hi,

zu 1) und 2) würde ich auch sagen, das man da nichts machen kann. Meines Wissens werden die Pumpen (Ladepumpe und Umwälzpumpe) von der Heizung nur Ein/Ausgeschaltet. Die Pumpe selbst, speziell die Umwälzpumpe kann dann natürlich eine variable Drehzahlsteuerung haben. Das ist aber dann eine unabhängige Funktion der Pumpe.
Es gibt auch noch eine optionale dritte Pumpe für die Warmwasserzirkulation. Da kann man in der Buderus einstellen wie oft die pro Stunde angehen soll. Das kann man auch per Parameter über die Software verstellen (zumindest bei meiner Implementierung).

Mit dem Mischer kenne ich mich nicht aus, ich habe selbst keinen verbaut.

Grüße vom anderen Sven

**Comment by qschneider** [comment]
Moin, Mischerstellung ist nur mit dem Mischermodul FM 241 aussagekräftig, welches dann als HK2 konfiguriert wird.
Die Mischerstellung wird aber komplett vom Modul in Abhängigkeit von VL Ist und Soll des HK2 gesteuert, ist also nur lesend.
Für HK1 ist er somit zu vernachlässigen.

Öldurchsatz ermittle ich angenähert wie folgt.
Öldurchsatz BE2.1 34 = 3,05 kg/h -> [Zusatzblatt Technische Daten 1999-11.pdf](https://github.com/the78mole/esphome_components/files/14066039/Zusatzblatt.Technische.Daten.1999-11.pdf)

Spezifische Gewicht von HE = 0,85 kg/l
3,05 kg/h ./. 0,85 kg/l = 3,588 l/h

Nachgemessen durch meinen Ölmengenzähler ergibt sich eher 3,63l/h aber das dann fast schon zu vernachlässigen.

Über die Brennerminuten kann man sich dann anhand des Faktors in Home Assistant (falls man es verwendet) einen kWh Zähler bauen, der dann im Energie Dashboard angezeigt werden kann.

`- sensor:
    - name: "Energieverbrauch Heizöl Heizung"
      unique_id: "heating_system_energy_est_1"
      unit_of_measurement: "kWh"
      device_class: energy
      state_class: total_increasing
      state: >
        {{ ( states('sensor.brennerlaufzeit_1') | float(0) / 60 * 36.3 ) | round(3) }}
      availability: >
        {{ states('sensor.brennerlaufzeit_1') | is_number }}`

**Reply by Kennny00** [reply]
Könnte ihr den Energieverbrauchzähler noch direkt in das Projekt einbinden?Am besten Einstellbar mit der Öldüse?

**Reply by qschneider** [reply]
Nutzt du nicht HA? Weil das obige ist imho besser dort aufgehoben und die Konfiguration hatte ich ja beschrieben.

**Reply by Kennny00** [reply]
Hi, ich benutze HA aber dachte es wäre sinnvoller alles zu bündeln aber wenn das nicht vorgesehen ist oder nachteilig ist werde ich den Sensor selbst anlegen. 
Danke trotzdem finde das Projekt super und bedanke mich sehr bei euch allen.
